src/main/java/duke/command/AddDeadline.java

```diff
@@ -0,0 +1,60 @@
+package duke.command;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.tasklist.Task;
+import duke.tasklist.TaskList;
+import duke.ui.UI;
+
+/**
+ * Command for adding a deadline task.
+ * This command parses the input and adds a new deadline task to the task list.
+ */
+public class AddDeadline extends Command {
+
+    /**
+     * Constructs an AddDeadline command with the given input string.
+     *
+     * @param s The input string containing the task description and deadline.
+     */
+    public AddDeadline(String s) {
```

Comment:
minor but perhaps can consider using the parameter name as 'string' for more clarity? that way later when u use it in methods such as execute it will be more clear